src/component/Legend.tsx

```diff
@@ -183,7 +191,7 @@ export class Legend extends PureComponent<Props, State> {
           this.wrapperNode = node;
         }}
       >
-        {renderContent(content, { ...this.props, payload: getUniqPayload(payload, payloadUniqBy, defaultUniqBy) })}
+        <PieContent content={content} props={this.props} />
```

Comment:
Each component renders its legend content differently (Area, Bar, Pie, all have different logic).

In this particular file I had to change `renderContent` into `PieContent` because I wanted to have a hook inside and hooks require functional components.

Reply:
Ah I'm going back through this one and I noticed it's in the `Legend.tsx` file. No this one will be just one `LegendContent`, my bad giving it a misleading name.

Reply:
No worries, sounds good